main/helpers.py

```diff
@@ -1,23 +1,50 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# In[3]:
+
+
 import pandas as pd
 import plotly.offline as py
 import plotly.graph_objs as go
 from .models import Tweet
 from django.utils.translation import gettext_lazy as _
 
 
-def df_from_tweets(only_symptoms=False):
+def df_all_tweets():
     dates = []
-    if only_symptoms == True:
-        iterator = Tweet.objects.filter(tweetannotation__symptom='yes')
-    else:
-        iterator = Tweet.objects.all()
+    iterator = Tweet.objects.all()
     for i in iterator:
         dates.append(i.date)
     df = pd.DataFrame(data={
         'date': dates,
         'has_symptom': dates})
     df['date'] = pd.to_datetime(df['date'])
-    adf = df.groupby(['date']).count()
+    df = df.groupby(['date']).count().reset_index()
+    return df
+   
+
+def compute_positive_rate():
+    df = pd.DataFrame.from_records(TweetAnnotation.objects.all().                                    values_list('tweet__tweet_id','tweet__date','symptom','uuid','created'),
+                                    columns=['tweet_id','date','symptom','uuid','timestamp'])
+    df=df[(df['symptom'].notna()) & (df['uuid'].notna())]
+    ids_yes = df.groupby(['tweet_id','date','symptom']).count()
+    ids_yes=pd.DataFrame(ids_yes['uuid'] / ids_yes.groupby('tweet_id')['uuid'].transform('sum')).reset_index()
+    ids_yes=ids_yes.loc[(ids_yes['symptom']=='yes')&(ids_yes['uuid']>=0.5),'id_str']
+    ids_yes=ids_yes.groupby('date').count().reset_index().rename(columns={'uuid':'nb_yes'})
+    df=df.groupby('date').count().reset_index().rename(columns={'uuid':'nb_annotations'})
+    df=pd.merge(df,ids_yes, on='date',how='outer')
+    df['percent_yes']=df['nb_yes']/df['nb_annotations']
+    df['date'] = pd.to_datetime(df['date'])
+    return df
+    
+def df_from_tweets(only_symptoms=False):
+    if only_symptoms == True:
+        df_yes=compute_positive_rate()
+        df_all=df_all_tweets()
+        df=pd.merge(df_all, df_yes, how='outer',on='date')
+        df['has_symptom']=df['percent_yes']*df['has_symptom']
+    adf=df
```

Comment:
if `only_symptoms` is `False`, then `df` doesn't exist and this fails as `variable referenced before assignment`